pkg/generate/app/pipeline.go

```diff
@@ -38,20 +39,20 @@ func NewImagePipeline(from string, image *ImageRef) (*Pipeline, error) {
 	}, nil
 }
 
-// NewBuildPipeline creates a new pipeline with components that are
-// expected to be built
+// NewBuildPipeline creates a new pipeline with components that are expected to
+// be built.
 func NewBuildPipeline(from string, input *ImageRef, outputDocker bool, strategy *BuildStrategyRef, env Environment, source *SourceRef) (*Pipeline, error) {
 	name, ok := NameSuggestions{source, input}.SuggestName()
 	if !ok {
-		name = fmt.Sprintf("app%d", rand.Intn(10000))
+		return nil, ErrNameRequired
 	}
+	name = namer.LimitLength(name, v2.RepositoryNameTotalLengthMax-kutil.DNS1123LabelMaxLength-1)
```

Comment:
Service names it's ok to mange - we shouldn't mangle other name types.

On Tue, Oct 13, 2015 at 4:54 AM, Rodolfo Carvalho <notifications@github.com>
wrote:

> In pkg/generate/app/pipeline.go
> <https://github.com/openshift/origin/pull/5082#discussion_r41841643>:
>
> >  	}
> > +	name = namer.LimitLength(name, v2.RepositoryNameTotalLengthMax-kutil.DNS1123LabelMaxLength-1)
>
> @smarterclayton <https://github.com/smarterclayton> I'm interpreting what
> you said about not making up names and getting rid of this code to mangle
> name to make it fit a certain length.
> This means that if name would be very long, instead of mangling it to some
> value the user did not explicitly provide, we'll error out saying the name
> is invalid (user can provide a valid name with --name or --to).
>
> @csrwng <https://github.com/csrwng> FYI
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/5082/files#r41841643>.
>